File: Behavioral/ChainOfResponsibility/JavaScript/supervisor.js
// Supervisor.js

const SupportHandler = require("./supportHandler");

class Supervisor extends SupportHandler {
  handleRequest(request) {
    if (request === "Intermediate Issue") {
      console.log(`Supervisor: Handling request - ${request}`);
    } else if (this.nextHandler) {
      this.nextHandler.handleRequest(request);
    } else {
      console.log(`No handler available for - ${request}`);
    }
  }
}

module.exports = Supervisor;

File: Behavioral/ChainOfResponsibility/CSharp/Supervisor.cs
// Supervisor.cs

using System;

public class Supervisor : SupportHandler {
  public override void HandleRequest(string request) {
    if (request == "Intermediate Issue") {
      Console.WriteLine($"Supervisor: Handling request - {request}");
    } else if (NextHandler != null) {
      NextHandler.HandleRequest(request); // Only pass if not handled
    } else {
      Console.WriteLine($"No handler available for - {request}");
    }
  }
}
